repos/system_upgrade/el7toel8/actors/migratentp/actor.py

```diff
@@ -0,0 +1,76 @@
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import call
+from leapp.models import FinalReport, NtpMigrationDecision
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+
+from subprocess import CalledProcessError
+from subprocess import Popen, PIPE
+
+class MigrateNtp(Actor):
+    name = 'migrate_ntp'
+    description = 'Migrate ntp and/or ntpdate configuration to chrony.'
+    consumes = (NtpMigrationDecision,)
+    produces = (FinalReport,)
+    tags = (ApplicationsPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        from leapp.libraries.actor import ntp2chrony
+
+        for decision in self.consume(NtpMigrationDecision):
+            if len(decision.migrate_services) == 0:
+                # Nothing to migrate
+                continue
+
+            # Enable the chronyd service (the chrony package should be
+            # installed after rpm upgrade)
+            try:
+                call(['systemctl', 'enable', 'chronyd.service'])
+            except CalledProcessError:
+                raise StopActorExecutionError('Could not enable chronyd service')
+
+            # Unpack archive with configuration files
+            #call(['/bin/sh', '-c', 'base64 -d | tar xv'], stdin=decision.config_tgz64)
+            p = Popen(['/bin/sh', '-c', 'base64 -d | tar xz -C /'], stdin=PIPE, stdout=None, stderr=None)
+            p.communicate(input=decision.config_tgz64.encode())
+
+            if 'ntpd' in decision.migrate_services:
+                ntp_conf = '/etc/ntp.conf'
+            else:
+                # Use a minimal secure ntp.conf if ntpd was disabled
+                ntp_conf = '/etc/ntp.conf.nosources'
+                with open(ntp_conf, 'w') as f:
+                    f.write('# This file was created to migrate ntpdate configuration to chrony\n')
+                    f.write('# without ntp configuration (ntpd service was disabled)\n')
+                    f.write('driftfile /var/lib/ntp/drift\n')
+                    f.write('restrict default ignore nomodify notrap nopeer noquery\n')
+
+            if 'ntpdate' in decision.migrate_services:
+                step_tickers = '/etc/ntp/step-tickers'
+            else:
+                step_tickers = ''
+
+            try:
+                ntp_configuration = ntp2chrony.NtpConfiguration('/', ntp_conf, step_tickers, 0)
+                ntp_configuration.write_chrony_configuration('/etc/chrony.conf', '/etc/chrony.keys',
+                                                             False, True)
+            except Exception as e:
+                raise StopActorExecutionError('ntp2chrony failed: {}'.format(e))
+
+            # Report ignored lines in ntp.conf, except 'disable monitor' from
+            # the default ntp.conf
+            for ignored_line in ntp_configuration.ignored_lines:
```

Comment:
Can't get the logic here. IIUC the first line that is ignored but isn't 'disable_monitor' will break the loop (not processing any other lines) and go straight to L72, while L66 will be reached if and only if ntp_configuration.ignored_lines == ['disable monitor'].
Is this intended? In that case - can you please rewrite it to make it more obvious?